### PR TITLE
build: download busted from own neovim/deps repository

### PR DESCRIPTION
**Pull request by dundargoc**
Downloading the necessary files all at once instead of doing dependency
handling with luarocks speeds up installation immensely. We speed up the
process even more by using luv as a replacement for the C modules in the
busted dependencies, which allows us to skip costly compilation times.

### CODE REVIEW

File: test/busted_runner.lua [hunk 1]
```diff
@@ -4,4 +4,61 @@ local suffix = (platform and platform.sysname:lower():find'windows') and '.dll'
 package.path = deps_install_dir.."/share/lua/5.1/?.lua;"..deps_install_dir.."/share/lua/5.1/?/init.lua;"..package.path
 package.cpath = deps_install_dir.."/lib/lua/5.1/?"..suffix..";"..package.cpath;
 
+local uv = vim.loop
```
Comment (clason) [inline test/busted_runner.lua:7]:
```suggestion
local uv = vim.uv
```

(unless we want to backport)

File: test/busted_runner.lua [hunk 2]
```diff
@@ -4,4 +4,61 @@ local suffix = (platform and platform.sysname:lower():find'windows') and '.dll'
...
+  local status, _ = pcall(uv.chdir, dir)
+  if status then
+    return true
+  else
+    return unpack(status)
+  end
+end
+
+function lfs.dir(path)
+  local fs = vim.loop.fs_scandir(path)
```
Comment (clason) [inline test/busted_runner.lua:55]:
```suggestion
  local fs = uv.fs_scandir(path)
```

File: test/busted_runner.lua [hunk 3]
```diff
@@ -4,4 +4,61 @@ local suffix = (platform and platform.sysname:lower():find'windows') and '.dll'
...
+  return function()
+    if not fs then
+      return
+    end
+    return vim.loop.fs_scandir_next(fs)
```
Comment (clason) [inline test/busted_runner.lua:60]:
```suggestion
    return uv.fs_scandir_next(fs)
```

**Review by clason — COMMENTED** [review]
looks like `busted.runner` is not loaded from `test` (not on `package.path`)?

File: test/busted_runner.lua [hunk 4]
```diff
@@ -4,4 +4,61 @@ local suffix = (platform and platform.sysname:lower():find'windows') and '.dll'
...
+function lfs.currentdir()
+  return uv.cwd()
+end
+
+function lfs.chdir(dir)
+  local status, _ = pcall(uv.chdir, dir)
+  if status then
+    return true
+  else
+    return unpack(status)
```
Comment (bfredl) [inline test/busted_runner.lua:50]: I fubbed this one, should've been `return nil, err`

**Review by bfredl — APPROVED** [review]

**Comment by justinmk** [comment]
Why do we pull this from https://github.com/neovim/deps instead of a release tarball from https://github.com/lunarmodules/busted ?

**Comment by dundargoc** [comment]
Because the release tarball doesn't include the lua files of all the dependencies from busted.

**Pull request merged**